lib/rules/no-else-return.js

```diff
@@ -80,15 +80,30 @@ module.exports = function(context) {
     /**
      * Check the consequent/body node to make sure it is not
      * a ReturnStatement or an IfStatement that returns on both
-     * code paths.  If it is, display the context report.
+     * code paths.
      *
      * @param {Node} node The consequent or body node
      * @param {Node} alternate The alternate node
-     * @returns {void}
+     * @returns {boolean} `true` if it is a Return/If node that always returns.
+     */
+    function checkForReturnOrIf(node) {
+        return checkForReturn(node) || checkForIf(node);
+    }
+
+
+    /**
+     * Check whether a node returns in every codepath.
+     * @param {Node} node The node to be checked
+     * @returns {boolean} `true` if it returns on every codepath.
      */
-    function checkForReturnOrIf(node, alternate) {
-        if (checkForReturn(node) || checkForIf(node)) {
-            displayReport(alternate);
+    function alwaysReturns(node) {
+        // If we have a BlockStatement, check each consequent body node.
+        if (node.type === "BlockStatement") {
+            return node.body.length > 0 && node.body.some(checkForReturnOrIf);
```

Comment:
`some` will return false for an empty array, so you don't need to check the length. Is there a way to optimize this check? Iterating the children of every IfStatement could get expensive.